src/app/Directives/authcheck.directive.ts

```diff
@@ -0,0 +1,29 @@
+import { Directive, OnInit, Input, ViewContainerRef, TemplateRef } from '@angular/core';
+import { AuthService } from '../services/auth.service';
+
+@Directive({
+  selector: '[appAuthcheck]'
+})
+export class AuthcheckDirective implements OnInit {
+
+  constructor(
+    private authService: AuthService,
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef
+  ) { }
+
+  condition: boolean;
```

Comment:
variable name should define the value it contains.
Try to make a input value for the directive.